liquidprompt

```diff
@@ -3835,6 +3849,113 @@ _lp_temperature_color() {
     lp_temperature_color="${LP_MARK_TEMP}${ret}${lp_temperature}°${NO_COL}"
 }
 
+######################
+# Current Disk Usage #
+######################
+
+# Convert bytes in human-readable, binary-prefixed notation (using powers of 2).
+# Precision can be 0, 1 or 2 digits. If not given, precision is two digits.
+__lp_bytes_to_human() { # bytes, [precision] -> lp_bytes, lp_bytes_units
+    local i="$1" prec="${2:-2}" size remainder=""
+    # This has to be split on multiple lines for Zsh 5.0 (see wiki)
+    local units=()
+    # Not much point in going past PiB, as Bash and Zsh start failing int math.
+    units=("B" "KiB" "MiB" "GiB" "TiB" "PiB" "EiB" "YiB" "ZiB")
+
+    for ((size=0; i > 1024 && size < ${#units[@]}-1; size++)); do
+        remainder=$((i % 1024 * 100 / 1024))
+        i=$((i / 1024))
+    done
+
+    if [[ -n $remainder ]]; then
+        if ((remainder < 10)); then
+            remainder="0$remainder"
+        fi
+    else
+        # Bytes case.
+        remainder="00"
+    fi
+
+    # Precision formating.
+    remainder="${remainder:0:$prec}"
+    if [[ -n $remainder ]]; then
+        remainder=".$remainder"
+    fi
+
+    lp_bytes="$i$remainder"
+    lp_bytes_units="${units[$size]}"
+}
+
+
+_lp_disk() {
+    ((LP_ENABLE_DISK)) || return 2
+
+    lp_disk=
+    lp_disk_human=
+    lp_disk_perc=
+
+    # df -P . prints only the current FS, in a portable format.
+    # -k to get values in kB (supposedly more portable).
+    # LANG is set to avoid locale mangling with the words number.
+    # Backslash avoids any alias of df.
+    local df=$(LANG=C \df -k -P . 2>/dev/null)
+    # Example output:
+    # Filesystem     1024-blocks      Used Available Capacity Mounted on
+    # /dev/nvme0n1p2   479608528 425188220  30011332      94% /
+
+    # Placeholder and capacity.
+    local _ capa
+    # Compute location of the targeted column on the safe line.
+    local short="${df#*1024-blocks}"
+    # Second line.
+    local second="${df#*$'\n'}"
+    # Consider only targeted columns.
+    # This may be extended to get other information if needed:
+    #       +- total
+    #       | +- used        +- mount point (may have spaces)
+    #       | |              |
+    #       v v              v
+    read -r _ _ lp_disk capa _ <<<"${second:${#df}-${#short}-11}"
+
+    # Without percent character.
+    lp_disk_perc=$(( 100 - ${capa:0:${#capa}-1} ))
+
+    [[ -z ${lp_disk} || -z ${lp_disk_perc} ]] && return 1
+
+    # Comparison in KiB.
+    if (( lp_disk <= LP_DISK_THRESHOLD || lp_disk_perc <= LP_DISK_THRESHOLD_PERC )); then
```

Comment:
>  ../liquidprompt: line 3926: LP_DISK_THRESHOLD_PERC: unbound variable

I think the problem is you never set it in the test script.

Reply:
Isn't the default value supposed to cover that kind of case?

Reply:
It would, yes. But the test doesn't load the default value, as it loads `liquidprompt` with the `--no-activate` flag. (Skipping activate is required, as otherwise LP would try to install itself, which would fail in tests.)

Arguably, we should make a way for tests to load the default values, but the tests often end up overriding them anyway to test functionality, so it never seemed worth it to me.